propagation/http_b3_propagator.go

```diff
@@ -184,12 +184,17 @@ func (b3 HTTPB3Propagator) extractSingleHeader(supplier apipropagation.Supplier)
 // extractTraceID parses the value of the X-B3-TraceId b3Header.
 func (b3 HTTPB3Propagator) extractTraceID(tid string) (traceID core.TraceID, ok bool) {
 	if hexStr32ByteRegex.MatchString(tid) {
-		traceID.High, _ = strconv.ParseUint(tid[0:(16)], 16, 64)
-		traceID.Low, _ = strconv.ParseUint(tid[(16):32], 16, 64)
-		ok = true
+		var err error
+		traceID, err = core.TraceIDFromHex(tid)
+		ok = err == nil
 	} else if b3.SingleHeader && hexStr16ByteRegex.MatchString(tid) {
-		traceID.Low, _ = strconv.ParseUint(tid[:16], 16, 64)
-		ok = true
+		b, err := hex.DecodeString(tid)
```

Comment:
@rghetia Here we are trying to parse a 16 character hex to a TraceID but that is not a valid trace-id, should we drop this out?

Reply:
This is checking if the hex encoded trace-id is 16 char long, not 16 byte.  It doesn't really matter if we are trying to parse from a single header or multiple headers, we should expect this to be a 32 long hex string.

Reply:
I've removed most of this function since this started to be handled by `core.TraceIDFromHex`

Reply:
yes, 16 char not 16 byte. According to [this](https://github.com/openzipkin/b3-propagation#traceid-1) it could be 32 or 16 hex lower character if there are multiple headers. Most of the deployments are probably using 32 hex char anyway. So I am fine with merging this.

Reply:
You're right! Sorry for the buzz, I got all caught up with w3c specs only. What do you think about us filling the first 16 char/ 8 bytes with random valuesas per [psec of w3c](https://www.w3.org/TR/trace-context/#trace-id)?
```
When a system operates with a trace-id that is shorter than 16 bytes, it SHOULD fill-in the extra bytes with random values rather than zeroes.

```

Reply:
if it is filled with random values then would backend be able to connect two spans? IMO. there are more chances that it would connect with zeroes than random values.
Not sure what others have to say.
Maybe we can open an issue, get opinions from others and make the change accordingly.

Reply:
👍  agree